packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -110,6 +110,10 @@ For example, to filter locales on only germany \`localeFilter: locale => locale.
 List of locales and their codes can be found in Contentful app -> Settings -> Locales`
         )
         .default(() => () => true),
+      typePrefix: Joi.string()
+        .allow(``) // By default, string doesn't allow empty string, so we need to allow it explicitly
```

Comment:
Should we allow empty strings though? Ideally there will always be a type prefix

Reply:
I think we should allow it, because it's a way to ensure the types match the ones in the Contentful API

Reply:
The issue I have with it is type names between plugins can clash fairly easily so omitting it can cause weird unexpected bugs due to content/cms changes without changing code. For less experienced users it can be hard to figure out what's going on when the api breaks with no code changes.

I think if we want to support some level of optional normalization between Gatsby types/shapes and those in CMS's it'd be better to add a helper that does it in the future

Reply:
For ex if Drupal has a `Person` type and Contentful also has one things will break

Reply:
Right, but Drupal should've always had type prefixes. The data layer consolidates multiple different data sources that aren't aware of each other so each source plugin should prefix their types. 
It's better to limit functionality until required vs opening it up in case someone might want it. 
To me the benefit vs risk ratio doesn't shake out here.

Benefit: someone one day might want contentful types to be closer to what's in their CMS
Risk: someone may set an empty string thinking it's a benefit and have conflicting types between other CMS's (for ex if they're also running the drupal version without type prefixes). In the case that this happens the errors likely wont be clear and it'll be frustrating to diagnose and fix.

For the benefit it's not clear who wants that, and if someone does want it it's not clear that removing the type prefix is the best way to achieve it. Other parts of the queried data shapes will be different than what's in Contentful anyway so imo it's better to be explicit that the user isn't dealing directly with the contentful API

Reply:
I've removed it, but I think we need a wider discussion about the principle of whether to allow empty prefixes in the plugins that don't already support it

Reply:
I definitely agree with that, thanks @ascorbic